src/main/java/mate/academy/dao/impl/CarDaoImplJdbc.java

```diff
@@ -0,0 +1,209 @@
+                + "AND cars.deleted = FALSE";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement(selectQuery)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseDataFromResultSet(resultSet));
+            }
+            setCars(cars);
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Something went wrong. Can't get car drivers by ID "
+                    + driverId, e);
+        }
+
+    }
+
+    private void setCars(List<Car> cars) {
+        for (Car car : cars) {
+            car.setDrivers(getCarDrivers(car.getId()));
+        }
+    }
+
+    private Car parseDataFromResultSet(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject("id", Long.class);
+        String model = resultSet.getString("model");
+        Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+        String name = resultSet.getString("name");
+        String country = resultSet.getString("country");
+        Manufacturer manufacturer = new Manufacturer(name, country);
+        manufacturer.setId(manufacturerId);
+        Car car = new Car(model, manufacturer);
+        car.setId(id);
+        return car;
+    }
+
+    private List<Driver> getCarDrivers(Long id) {
+        String query = "SELECT drivers.name, drivers.license_number, "
+                + "cars_drivers.driver_id FROM cars_drivers "
+                + "JOIN drivers ON cars_drivers.driver_id = drivers.id "
+                + "WHERE cars_drivers.car_id = ?";
+        List<Driver> drivers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                String name = resultSet.getString("name");
+                String licenceNumber = resultSet.getString("license_number");
+                Long driverId = resultSet.getObject("driver_id", Long.class);
+                Driver driver = new Driver(name, licenceNumber);
+                driver.setId(driverId);
+                drivers.add(driver);
+            }
+            return drivers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Something went wrong. Can't get drivers from car "
+                    + id, e);
+        }
+    }
+
+    private void insertDrivers(Car car) {
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
```

Comment:
Open one Connection in another Connection it's bad practice, try to use just one connection, instead of this you can create another PreparedStatement. Write like second argument in method Connection connection
```suggestion
        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
```

Reply:
fixed